core/new-gui/src/app/workspace/service/validation/validation-workflow.service.ts

```diff
@@ -0,0 +1,141 @@
+import { OperatorSchema} from '../../types/operator-schema.interface';
+import { WorkflowActionService } from './../workflow-graph/model/workflow-action.service';
+import * as Ajv from 'ajv';
+import { isEqual } from 'lodash-es';
+@Injectable()
+export class ValidationWorkflowService {
+  private operatorSchemaList: ReadonlyArray<OperatorSchema> = [];
+  private readonly operatorValidationStream =  new Subject <{status: boolean, operatorID: string}>();
+
+  /**
+   * subcribe the add opertor event, delete operator event, add link event, delete link event
+   * and change operator property event. observe each change and record changes in operatorValidationStream
+   * @param texeraGraph
+   * @param workflowActionService
+   */
+  constructor(private operatorMetadataService: OperatorMetadataService,
+    private workflowActionService: WorkflowActionService) {
+    this.operatorMetadataService.getOperatorMetadata().subscribe(
+        value => { this.operatorSchemaList = value.operators; }
+      );
+
+    this.workflowActionService.getTexeraGraph().getOperatorAddStream()
+      .subscribe(
+        value => {
+          const operatorID = value.operatorID;
+          const status = this.validateOperator(operatorID);
+          this.operatorValidationStream.next({status, operatorID});
+          }
+        );
+
+    this.workflowActionService.getTexeraGraph().getOperatorDeleteStream()
+      .subscribe(
+        value => {
+          const operatorID = value.deletedOperator.operatorID;
+          const status = this.validateOperator(operatorID);
+          this.operatorValidationStream.next({status, operatorID});
+         }
+        );
+
+    this.workflowActionService.getTexeraGraph().getLinkAddStream()
+      .subscribe(value => {
+        let operatorID = value.source.operatorID;
+        let status = this.validateOperator(operatorID);
+        this.operatorValidationStream.next({status, operatorID});
+        operatorID = value.target.operatorID;
+        status = this.validateOperator(operatorID);
+        this.operatorValidationStream.next({status, operatorID});
+      });
+
+
+    this.workflowActionService.getTexeraGraph().getLinkDeleteStream()
+      .subscribe(value => {
+        let operatorID = value.deletedLink.source.operatorID;
+        let status = this.validateOperator(operatorID);
+        this.operatorValidationStream.next({status, operatorID});
+        operatorID = value.deletedLink.target.operatorID;
+        status = this.validateOperator(operatorID);
+        this.operatorValidationStream.next({status, operatorID});
+      });
+
+    this.workflowActionService.getTexeraGraph().getOperatorPropertyChangeStream()
+      .subscribe(value => {
+        const operatorID = value.operator.operatorID;
+        const status = this.validateOperator(operatorID);
+        this.operatorValidationStream.next({status, operatorID});
+      });
```

Comment:
For each of the handler for each observable stream, write some comments to describe what each handler does.